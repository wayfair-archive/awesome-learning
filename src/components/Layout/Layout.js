import React from "react";
import PropTypes from 'prop-types';
import Helmet from "react-helmet";
import cx from "classnames";
import Header from "../Header";

import "./Layout.scss";

const Layout = ({ children, title, description, isFullBleed }) => {
  const layoutClass = cx("Layout", {
    "Layout--fullBleed": isFullBleed
  });
  return (
    <>
      <Helmet>
        <html lang="en" />
        <title>{title}</title>
        <meta name="description" content={description} />
        <link
          href="https://fonts.googleapis.com/css?family=Raleway:200,400"
          rel="stylesheet"
        />
      </Helmet>
      <Header />
      <div className={layoutClass}>{children}</div>
    </>
  );
};

Layout.defaultProps = {
  isFullBleed: false
};

Layout.propTypes = {
  children: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  isFullBleed: PropTypes.string,
}
export default Layout;
