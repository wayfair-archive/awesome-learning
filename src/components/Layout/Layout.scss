.Layout {
  max-width: $layout-width;
  padding: 0 1.5rem;
  margin: 0 auto;
  &--fullBleed {
    padding: 0;
    max-width: 100%;
  }
}
