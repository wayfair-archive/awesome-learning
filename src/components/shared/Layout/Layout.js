import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import cx from 'classnames';
import Header from '../../Header';
import LastLessonProvider from '../../../providers/LastLessonProvider';
import './Layout.scss';

const Layout = ({children, title, description, isFullBleed}) => {
  const layoutClass = cx('Layout', {
    'Layout--fullBleed': isFullBleed
  });
  return (
    <>
      <Helmet>
        <html lang="en" />
        <title>{title}</title>
        <meta name="description" content={description} />
        <link
          href="https://fonts.googleapis.com/css?family=Raleway:200,400"
          rel="stylesheet"
        />
      </Helmet>
      <Header />
      <div className={layoutClass}>
        <LastLessonProvider>{children}</LastLessonProvider>
      </div>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  isFullBleed: PropTypes.bool
};

Layout.defaultProps = {
  isFullBleed: false
};

export default Layout;
