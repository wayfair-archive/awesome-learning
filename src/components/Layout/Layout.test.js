import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';

describe('Layout', () => {
  const props = {
    children: 'test',
    description: 'test',
    title: 'test'
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Layout {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
