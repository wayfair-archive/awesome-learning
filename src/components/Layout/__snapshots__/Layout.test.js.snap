// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <StaticQuery
    render={[Function]}
  />,
  <div
    className="Layout"
  >
    test
  </div>,
]
`;
